apis/Google.Cloud.Diagnostics.AspNetCore/Google.Cloud.Diagnostics.AspNetCore/Trace/ContextManagedTracer.cs

```diff
@@ -0,0 +1,77 @@
+﻿// Copyright 2017 Google Inc. All Rights Reserved.
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Threading.Tasks;
+using Google.Cloud.Diagnostics.Common;
+using Microsoft.AspNetCore.Http;
+using Google.Api.Gax;
+
+namespace Google.Cloud.Diagnostics.AspNetCore
+{
+    /// <summary>
+    /// An <see cref="IManagedTracer"/> that will pull the the current tracer from
+    /// the <see cref="HttpContext"/>. The current tracer is set per request in the
+    /// <see cref="CloudTraceMiddleware"/> via the <see cref="ContextTracerManager"/>.
+    /// This allows for this to be a singleton <see cref="IManagedTracer"/> so users
+    /// do not need to worry about scoping of the tracer.
+    /// </summary>
+    internal sealed class ContextManagedTracer : IManagedTracer
```

Comment:
If we find we want a similar class elsewhere, we could have a DelegatingTracer which accepted a `Func<IManagedTracer>`, and made each call just delegate to that and then pass the arguments to it, abstracting the `ContextTracerManager.GetCurrentTracer(_accessor)` part.

Do we already have anything like this in the AspNet code?